samples/dotnet/kernel-syntax-examples/Example12_Planning.cs

```diff
@@ -259,55 +181,51 @@ private static IKernel InitializeKernelAndPlanner(out IDictionary<string, ISKFun
             Env.Var("AZURE_OPENAI_ENDPOINT"),
             Env.Var("AZURE_OPENAI_KEY"));
 
-        // Load native skill into the kernel skill collection, sharing its functions with prompt templates
-        planner = kernel.ImportSkill(new PlannerSkill(kernel, maxTokens), "planning");
+        planner = new SequentialPlanner(kernel, new PlannerConfig() { MaxTokens = maxTokens });
+
         return kernel;
     }
 
-    private static async Task<SKContext> ExecutePlanAsync(
+    private static async Task<Plan> ExecutePlanAsync(
         IKernel kernel,
-        IDictionary<string, ISKFunction> planner,
-        SKContext executionResults,
+        Plan inputPlan,
+        string input = "",
         int maxSteps = 10)
     {
         Stopwatch sw = new();
         sw.Start();
 
         // loop until complete or at most N steps
-        for (int step = 1; !executionResults.Variables.ToPlan().IsComplete && step < maxSteps; step++)
+        var plan = inputPlan;
+        for (int step = 1; plan.HasNextStep && step < maxSteps; step++)
         {
-            var results = await kernel.RunAsync(executionResults.Variables, planner["ExecutePlan"]);
-            if (results.Variables.ToPlan().IsSuccessful)
+            try
             {
-                Console.WriteLine($"Step {step} - Execution results:");
-                Console.WriteLine(results.Variables.ToPlan().PlanString);
-
-                if (results.Variables.ToPlan().IsComplete)
-                {
-                    Console.WriteLine($"Step {step} - COMPLETE!");
-                    Console.WriteLine(results.Variables.ToPlan().Result);
-
-                    // Console.WriteLine("VARIABLES: ");
-                    // Console.WriteLine(string.Join("\n\n", results.Variables.Select(v => $"{v.Key} = {v.Value}")));
-                    break;
-                }
-
-                // Console.WriteLine($"Step {step} - Results so far:");
-                // Console.WriteLine(results.ToPlan().Result);
+                plan = await kernel.StepAsync(input, plan);
```

Comment:
I would change this to:
```
    inputPlan.InvokeNextStepAsync();
```

Since we are breaking on the first step error would move the try out of the for, this would break as a bonus.

Reply:
Makes sense on try. Could update to something like this to show options:

```
if (string.IsNullOrEmpty(input))
{
    await plan.InvokeNextStepAsync(kernel.CreateNewContext());
    // or await kernel.StepAsync(plan);
}
else
{
    plan = await kernel.StepAsync(input, plan);
}
```